### PR TITLE
[SRVKE-1152] Land new java kafka source

### PR DESCRIPTION
**Pull request by matzew**
https://issues.redhat.com/browse/SRVKE-1152

PR introduces Java-based kafka source from `broker` repo, and a migration job.

Based on:
* https://github.com/openshift-knative/serverless-operator/pull/1393
* https://github.com/openshift-knative/serverless-operator/pull/1413
* https://github.com/openshift-knative/serverless-operator/pull/1417

Summary, with a working 1.1.x of Knative Eventing, we move on to _replace_ the source implementation :tada:

### CODE REVIEW

File: knative-operator/pkg/controller/knativekafka/knativekafka_controller_test.go [hunk 1]
```diff
@@ -51,7 +51,8 @@ func TestKnativeKafkaReconcile(t *testing.T) {
 		instance: makeCr(withChannelEnabled, withSourceEnabled),
 		exists: []types.NamespacedName{
 			{Name: "kafka-ch-controller", Namespace: "knative-eventing"},
-			{Name: "kafka-controller-manager", Namespace: "knative-eventing"},
+			// TODO: commented out, since rbac injection is missing
```
Comment (matzew) [inline knative-operator/pkg/controller/knativekafka/knativekafka_controller_test.go:54]: @pierDipi I left them out, on purpose - also I saw your PR to bring this in. Let's land yours after this

**Comment by matzew** [comment]
/hold
wait until all passed

**Comment by matzew** [comment]
@pierDipi 
```
    source_native_kafka_broker_ksvc_test.go:125: ConfigMap still contains the finalizer kafka.brokers.eventing.knative.dev/serverless-tests-smoke-test-native-kafka-broker
```

didn't we fix this, but than forgot to backport? :thinking:

**Comment by matzew** [comment]
https://github.com/knative-sandbox/eventing-kafka-broker/pull/1917

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**